GitCommands/Submodules/SubmoduleStatusProvider.cs

```diff
@@ -326,6 +338,24 @@ private async Task UpdateSubmodulesStatusAsync(GitModule module, [CanBeNull] IRe
             }
         }
 
+        /// <summary>
+        /// Set the top module as dirty
+        /// If any module is changed
+        /// </summary>
+        /// <param name="topModuleWorkingDir">path to top module</param>
+        private void SetTopModuleAsDirty(string topModuleWorkingDir)
+        {
+            if (_submoduleInfos[topModuleWorkingDir].Detailed == null)
```

Comment:
I missed this on the first review. Are you sure `_submoduleInfos[topModuleWorkingDir]` is safe?

Reply:
Yes, added on line 111